Source/WebCore/platform/graphics/skia/GraphicsContextSkia.h

```diff
@@ -27,25 +27,36 @@
 
 #if USE(SKIA)
 
+#include "GLFence.h"
 #include "GraphicsContext.h"
 WTF_IGNORE_WARNINGS_IN_THIRD_PARTY_CODE_BEGIN
 #include <skia/core/SkCanvas.h>
+#include <skia/core/SkImage.h>
 #include <skia/effects/SkDashPathEffect.h>
 WTF_IGNORE_WARNINGS_IN_THIRD_PARTY_CODE_END
 #include <wtf/CompletionHandler.h>
+#include <wtf/HashMap.h>
+
+class SkSurface;
 
 namespace WebCore {
 
 class WEBCORE_EXPORT GraphicsContextSkia final : public GraphicsContext {
 public:
-    GraphicsContextSkia(SkCanvas&, RenderingMode, RenderingPurpose, CompletionHandler<void()>&& = nullptr);
+    enum class ContextType {
+        RegularContext,
+        RecordingContext
```

Comment:
Maybe we could add a RenderingPurpose so that we don't need to add another parameter

Reply:
No longer a parameter, just a private `enum class ContextMode { PaintingMode, RecordingMode }`. beginRecording() switches to RecordingMode, endRecording() back to PaintingMode. No more consturctor parameter, good suggestion!